src/graphql/root/query/btc-price-list.ts

```diff
@@ -35,39 +46,32 @@ const BtcPriceListQuery = GT.Field({
       type: GT.NonNull(PriceGraphRange),
     },
   },
-  resolve: async (_, args, { logger }) => {
-    const { range } = args
-
-    const rangeStart = rangeStartDateMS(range)
-
-    if (!rangeStart) {
-      throw new Error("Invalid range")
-    }
+  resolve: async (_, args) => {
+    const range = parseRange(args.range)
+    const interval = parseInterval(args.range)
 
-    const hourlyPrices = await getHourlyPrice({ logger })
+    if (!range) throw new Error("Invalid range")
 
-    const prices: PricePointType[] = []
+    const hourlyPrices = await Prices.getPriceHistory({ range, interval })
+    if (hourlyPrices instanceof Error) throw hourlyPrices
 
-    for (const price of hourlyPrices.reverse()) {
-      if (1000 * price.id < rangeStart) {
-        break
-      }
-      const btcPriceInCents = price.o * 100 * 10 ** 8
-      prices.unshift({
-        timestamp: price.id,
+    const prices: PricePointType[] = hourlyPrices.map(({ date, price }) => {
+      const btcPriceInCents = price * 100 * 10 ** 8
```

Comment:
Should we make `10 ** 8` a constant like we have with different millisecond values (maybe `SATS_PER_BTC`)?

We also use `Math.pow(10, 8)` in other places.

Reply:
Fixed